contributingGuides/HYBRID_APP.md

```diff
@@ -0,0 +1,212 @@
+### Should I rebuild HybridApp after bumping a `node_modules` library?
+The `package-lock.json` file contains information about exact versions of `node_modules` that will be installed on your machine. If you've bumped a dependency on your PR you can easily see if you would need to rebuild the app. In this case go to the `./node_modules/<PACKAGE_NAME>`, and see if there are any Objective-C, Swift, Java, Kotlin or C++ files. Usually they are located in `ios` or `android` folders.
+### How to clear platform-specific cache?
+Executing `npm run clean` clears cache for React Native, Android, and iOS. It means that the whole process may take a while, even though you try to build only one platform. In this case you case you can pass additional arguments to specify which cache should be cleared in order to save some time:
+- **React Native:** `npm run clean -- --react-native` 
+	- `npm`, `node_modules`, `watchman`, Metro
+- **Android:** `npm run clean -- --android` 
+	- `build`, `.cxx`, `.gradle`
+- **iOS:** `npm run clean -- --ios`
+	- `Pods`, `DerivedData`, `build`, `cocoapods`
+- **YAPL**: `npm run clean -- --npm`
+	- `npm`, `Mobile-Expensify/node_modules` (OldDot-specific)
+## Android
+### Error: `Could not find method autolinkLibrariesFromCommandForPath()`
+
+This is an error that indicates, that patches from `./Mobile-Expensify/patches` were not applied. In this case:
+1. open a new terminal
+2. run `rm -rf node_modules && npm i`
+3. run `npm run android`
+### Error: `A problem occurred starting process 'command 'node''`
+
+Android Studio for some reason sometimes is unable to find `node`, even though its terminal has correct access to all environment variables. According to StackOverflow's threads the problem started appearing in 2021. In some cases linking node worked (`sudo ln -s "$(which node)" /usr/local/bin/node`), but the easiest way is to open Android Studio directly from terminal:
+1. close Android Studio
+2. run `open -a /Applications/Android\ Studio.app`
+### Error: `> Task :installDebug FAILED`
+
+Actually you should be happy whenever you see this error - it means that the build failed, but the `.apk` hasn't been installed on your device/emulator. In order to fix this, refer to: [[#How to find an `.apk`, and install it on your device?]]
+
+IMPORTANT: It's easily to confuse this error with a very similar one: `Failed to install the app. Command failed with exit code 1: ./gradlew installDebug`. In fact, this one may have many possible reasons, and in order to debug it correctly, you should look for a Gradle **Task** that failed. 
+### Error: `undefined is not an object (evaluating 'Store.ReportHistory.bindCacheClearingEvents')`
+This error indicates that YAPL JS (OldDot's JavaScript code) hasn't been built properly. In order to fix that, do the following:
+1. cd to `Mobile-Expensify`
+2. run `npm run grunt:build:shared`
```

Comment:
This should be fixed, but lets keep it just in case it breaks again 🤷

Reply:
Yeah, agreed, just in case it appears for some strange reason